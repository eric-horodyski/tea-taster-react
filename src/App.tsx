import { Redirect, Route } from 'react-router-dom';
import { IonApp, IonRouterOutlet, setupIonicReact } from '@ionic/react';
import { IonReactRouter } from '@ionic/react-router';

import SplashContainer from './components/splash/SplashContainer';
import AuthProvider from './providers/AuthProvider';
import TeaProvider from './providers/TeaProvider';

import { PrivateRoute } from './routes/PrivateRoute';
import Tabs from './routes/Tabs';
import LoginPage from './pages/login/LoginPage';
import UnlockPage from './pages/unlock/UnlockPage';
import StartPage from './pages/StartPage';

/* Core CSS required for Ionic components to work properly */
import '@ionic/react/css/core.css';

/* Basic CSS for apps built with Ionic */
import '@ionic/react/css/normalize.css';
import '@ionic/react/css/structure.css';
import '@ionic/react/css/typography.css';

/* Optional CSS utils that can be commented out */
import '@ionic/react/css/padding.css';
import '@ionic/react/css/float-elements.css';
import '@ionic/react/css/text-alignment.css';
import '@ionic/react/css/text-transformation.css';
import '@ionic/react/css/flex-utils.css';
import '@ionic/react/css/display.css';

/* Theme variables */
import './theme/variables.css';
import './theme/global.css';
import { hideContentsInBackground, isHidingContentsInBackground } from './utils/device';

setupIonicReact();

isHidingContentsInBackground().then((hide) => hideContentsInBackground(hide));

const App: React.FC = () => (
  <IonApp>
    <IonReactRouter>
      <AuthProvider>
        <SplashContainer>
          <IonRouterOutlet>
            <Route exact path="/unlock">
              <UnlockPage />
            </Route>
            <Route exact path="/login">
              <LoginPage />
            </Route>
            <Route path="/tabs">
              <PrivateRoute>
                <TeaProvider>
                  <Tabs />
                </TeaProvider>
              </PrivateRoute>
            </Route>
            <Route exact path="/">
              <StartPage />
            </Route>
          </IonRouterOutlet>
        </SplashContainer>
      </AuthProvider>
    </IonReactRouter>
  </IonApp>
);

export default App;
