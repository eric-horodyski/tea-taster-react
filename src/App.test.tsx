import { render } from '@testing-library/react';
import App from './App';

describe('<App />', () => {
  it('renders without crashing', async () => {
    const { baseElement } = render(<App />);
    expect(baseElement).toBeDefined();
  });

  it('renders consistently', async () => {
    const { asFragment } = render(<App />);
    expect(asFragment()).toMatchSnapshot();
  });
});
