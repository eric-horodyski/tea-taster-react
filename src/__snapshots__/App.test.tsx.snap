// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<App /> > renders consistently 1`] = `
<DocumentFragment>
  <ion-app>
    <ion-router-outlet>
      <div
        class="ion-page ion-page-invisible"
      >
        <ion-header>
          <!---->
          <ion-toolbar>
            <ion-title>
              Blank
            </ion-title>
          </ion-toolbar>
        </ion-header>
        <ion-content
          fullscreen="true"
        >
          <ion-header
            collapse="condense"
          >
            <!---->
            <ion-toolbar>
              <ion-title
                size="large"
              >
                Blank
              </ion-title>
            </ion-toolbar>
          </ion-header>
          <div
            class="container"
          >
            <strong>
              Ready to create an app?
            </strong>
            <p>
              Start with Ionic 
              <a
                href="https://ionicframework.com/docs/components"
                rel="noopener noreferrer"
                target="_blank"
              >
                UI Components
              </a>
            </p>
          </div>
        </ion-content>
      </div>
    </ion-router-outlet>
  </ion-app>
</DocumentFragment>
`;
